### PR TITLE
Added support for decimal timezones

### LINKED ISSUE #51: [Issue]: User playback report not working.
### Please describe your bug

User playback not working.
{timezoneOffset: ["The value '5.5' is not valid."]}

### Jellyfin Version

10.7.7

### if other:

_No response_

### Environment

```markdown
- OS:Ubuntu
- Virtualization: Docker
- Browser: Chrome
```

### Jellyfin logs

_No response_

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

errors: {timezoneOffset: ["The value '5.5' is not valid."]}
timezoneOffset: ["The value '5.5' is not valid."]
0: "The value '5.5' is not valid."
status: 400
title: "One or more validation errors occurred."
traceId: "00-dc0f5f85f64e744c83a3e704e44d04d2-3e7269ad47648740-00"
type: "https://tools.ietf.org/html/rfc7231#section-6.5.1"

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by theKAKAN**
Fixes #51

### CODE REVIEW

**Comment by h1dden-da3m0n** [comment]
TIL that there are .5 time zones !?

File: Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs [hunk 1]
```diff
@@ -464,7 +469,12 @@ public void UpdatePlaybackAction(PlaybackInfo playInfo)
             sql_query += "AND UserId not IN (select UserId from UserList) ";
             sql_query += "GROUP BY UserId, date ORDER BY UserId, date ASC";
 
-            endDate = endDate.AddHours(-timezoneOffset);
+            // Convert the decimal to minutes
+            // 0.5*60 = 30
+            int minOffset = (int)(( timezoneOffset < 0.0 ? (-timezoneOffset)%1 : timezoneOffset%1 )*60);
+            // Turn to negative in case timezoneOffset is negative
+            minOffset = ( timezoneOffset < 0.0 ? -minOffset : minOffset );
+            endDate = endDate.AddHours((int)-timezoneOffset).AddMinutes(-minOffset);
```
Comment (h1dden-da3m0n) [inline Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs:477]: since this code seems to be repeated 6, would it not call for a function to then call 6 times instead?

File: Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs [hunk 2]
```diff
@@ -413,7 +413,12 @@ public void UpdatePlaybackAction(PlaybackInfo playInfo)
             {
                 using var connection = CreateConnection(true);
                 using var statement = connection.PrepareStatement(sql_query);
-                var fromDate = DateTime.Parse(date + " 00:00:00").AddHours(-timezoneOffset);
+                // Convert the decimal to minutes
+                // 0.5*60 = 30
+                int minOffset = (int)(( timezoneOffset < 0.0 ? (-timezoneOffset)%1 : timezoneOffset%1 )*60);
+                // Turn to negative in case timezoneOffset is negative
+                minOffset = ( timezoneOffset < 0.0 ? -minOffset : minOffset );
+                var fromDate = DateTime.Parse(date + " 00:00:00").AddHours((int)-timezoneOffset).AddMinutes(-minOffset);
```
Comment (h1dden-da3m0n) [inline Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs:421]: if you extract the other 6 occurrences into a func and craft it right you could also use it here I would wager

File: Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs (hunk 1 above)
Comment (theKAKAN) [reply Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs:477]: Oof, I'm sorry. I should have lol

I just wanted the plugin to work instead of thinking.
Honestly, there's probably an even better way to do it. I'll try when I get some time today. Thank you!

File: Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs (hunk 2 above)
Comment (theKAKAN) [reply Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs]: Could you take a look now?

**Comment by theKAKAN** [comment]
> TIL that there are .5 time zones !?

Those are basically the 30 minute additions (_30/60 is .5, after all_).
So, GMT +1:30 is represented as 1.5 and so on.

File: Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs [hunk 3]
```diff
@@ -514,7 +514,7 @@ public void UpdatePlaybackAction(PlaybackInfo playInfo)
             }
 
             SortedDictionary<string, int> report_data = new SortedDictionary<string, int>();
-
+
```
Comment (h1dden-da3m0n) [inline Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs:517]:
```suggestion

```

unnecessary whitespace change

**Review by h1dden-da3m0n — APPROVED** [review]
LGTM

I assume you have tested this with your setup and TZ :wink:
Only outstanding bit is the obsolete whitespace change, otherwise I will merge this tomorrow.

*I dint know that `addHours` could work with `double` either :sweat_smile:
